### PR TITLE
ci: :white_check_mark: bring docker build for release

### LINKED ISSUE #33: change pipeline to adapt github release tag

### PR DESCRIPTION
**Pull request by akakshuki**

### CODE REVIEW

File: .github/workflows/docker-build.yml [hunk 1]
```diff
@@ -1,38 +1,56 @@
...
+            echo DOCKER_IMAGE="ghcr.io/planthor/planthoridentityserver:$DOCKER_TAG" >> "$GITHUB_OUTPUT"
+          elif [[ $GITHUB_REF =~ ^refs/tags/* ]]; then
+            DOCKER_TAG="${{  github.ref_name }}"
+            echo DOCKER_IMAGE="ghcr.io/planthor/planthoridentityserver:$DOCKER_TAG" >> "$GITHUB_OUTPUT"
+          else 
+            DOCKER_TAG="${{ github.head_ref }}"
+            echo DOCKER_IMAGE="planthordev/identityserver:$DOCKER_TAG" >> "$GITHUB_OUTPUT"
+          fi
 
       - name: Build and push Docker images
```
Comment (zovippro1996) [inline .github/workflows/docker-build.yml:54]: Does this mean we will create new image and push every time we create a PR, even though it is not merged? 
If that is the case, then this Github Action should be run on Push only.

File: .github/workflows/docker-build.yml [hunk 2]
```diff
@@ -1,38 +1,56 @@
...
         id: define_docker_image
         run: |
-          DOCKER_TAG="testcandidates-$(date +%Y%m%d%H%M%S)"
-          echo DOCKER_IMAGE="planthordev/identityserver:$DOCKER_TAG" >> "$GITHUB_OUTPUT"
+          DOCKER_TAG=""
+          if [[ $GITHUB_REF == 'refs/heads/main' ]]; then
+            DOCKER_TAG="latest"
+            echo DOCKER_IMAGE="ghcr.io/planthor/planthoridentityserver:$DOCKER_TAG" >> "$GITHUB_OUTPUT"
+          elif [[ $GITHUB_REF =~ ^refs/tags/* ]]; then
+            DOCKER_TAG="${{  github.ref_name }}"
```
Comment (zovippro1996) [inline .github/workflows/docker-build.yml:45]: spacing redundant.
```suggestion
            DOCKER_TAG="${{ github.ref_name }}"
```

File: .github/workflows/docker-build.yml (hunk 1 above)
Comment (akakshuki) [reply .github/workflows/docker-build.yml:54]: In this case, I would like to create an image for each branch to have a pull request. The developer needs to test their feature as container. I think its okay then we could have a clean house keeping job in the future

File: .github/workflows/docker-build.yml [hunk 3]
```diff
@@ -1,38 +1,56 @@
...
-
   docker-build:
     runs-on: ubuntu-latest
-    name: Run docker build  
+    name: Run docker build
     steps:
-      
       - uses: actions/checkout@v4
 
       - name: Docker Login
```
Comment (zovippro1996) [inline .github/workflows/docker-build.yml:22]: Are these 2 redundant? Why we need to login 2 times.

File: .github/workflows/docker-build.yml (hunk 1 above)
Comment (zovippro1996) [reply .github/workflows/docker-build.yml:54]: Then we should create image and push to registry for "on push" only, we don't have to create image and push with "on pull_request" right ?

File: .github/workflows/docker-build.yml (hunk 3 above)
Comment (akakshuki) [reply .github/workflows/docker-build.yml:22]: I need this login to push images to github registry

File: .github/workflows/docker-build.yml (hunk 1 above)
Comment (akakshuki) [reply .github/workflows/docker-build.yml:54]: Okay will remove feature branche case

**Review by zovippro1996 — APPROVED** [review]

**Pull request merged**